### PR TITLE
Add integration tests for error messages from balance endpoint

### PR DESCRIPTION
**Pull request by piotr-iohk**
### Issue Number

ADP-1241
ADP-1225

### Comments

This PR adds:
- integration test coverage for the numeric/variable parts of a collateral selection error.
- integration test coverage for error messages from the balance point. (These are currently marked as pending, as we expect to improve these errors.)

### CODE REVIEW

File: lib/core-integration/src/Test/Integration/Scenario/API/Shelley/TransactionsNew.hs [hunk 1]
```diff
@@ -1194,6 +1195,135 @@ spec = describe "NEW_SHELLEY_TRANSACTIONS" $ do
...
+        wa <- fixtureWalletWith @n ctx [2 * 1_000_000]
+        let balancePayload = Json PlutusScenario.pingPong_1
+        rTx <- request @ApiSerialisedTransaction ctx
+            (Link.balanceTransaction @'Shelley wa) Default balancePayload
+        verify rTx
+            [ expectResponseCode HTTP.status403
+            , expectErrorMessage errMsg403Fee
+            ]
+
+    it "TRANS_NEW_BALANCE_02c - Cannot balance on when I cannot afford collateral" $ \ctx -> runResourceT $ do
```
Comment (piotr-iohk) [inline lib/core-integration/src/Test/Integration/Scenario/API/Shelley/TransactionsNew.hs]: Possibly ADP-1241 case @jonathanknowles.

Comment (jonathanknowles) [reply lib/core-integration/src/Test/Integration/Scenario/API/Shelley/TransactionsNew.hs]: @piotr-iohk I've prepared some enhancements to this test in the following branch: 
https://github.com/input-output-hk/cardano-wallet/compare/jonathanknowles/adp-1225/test-enhancements

Will make this into a PR tomorrow morning!

**Review by jonathanknowles — APPROVED** [review]

File: lib/core-integration/src/Test/Integration/Scenario/API/Shelley/TransactionsNew.hs (hunk 1 above)
Comment (jonathanknowles) [reply lib/core-integration/src/Test/Integration/Scenario/API/Shelley/TransactionsNew.hs]: Update, I've added these commits to this PR.

**Pull request merged**